File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.rb
=begin
 Algoritmo da Estrutura de dados linear estática (não dinâmica) de Pesquisa/Busca 
 buscaSequencial para pesquisa sequencial sem a utilização de sentinela
=end

def busca_sequencial(array, valor_procurado)
    for i in 0...array.length
        if array[i] == valor_procurado
            return i  # Retorna o índice do valor procurado
        end
    end
    return -1  # Retorna -1 se o valor não foi encontrado
end

array = [10, 5, 2, 8, 3, 7, 1, 6, 9, 4]
valor_procurado = 7

indice_encontrado = busca_sequencial(array, valor_procurado)

if indice_encontrado != -1
    puts "Valor #{valor_procurado} encontrado no índice: #{indice_encontrado}"
else
    puts "Valor #{valor_procurado} não encontrado no array."
end

File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.cpp
/*
♦ Algoritmo 09: Estrutura de Dados Linear Estática (não Dinâmica) de Pesquisa/Busca Pesquisa Sequencial (sem a utilização de variável/sentinela).
*/

#include <iostream>
using namespace std;

int buscaSequencial(int array[], int tamanho, int valorProcurado) {
    for (int i = 0; i < tamanho; i++) {
        if (array[i] == valorProcurado) {
            return i; // Retorna o índice do valor procurado
        }
    }
    return -1; // Retorna -1 se o valor não foi encontrado
}

int main() {
    int array[] = {10, 5, 2, 8, 3, 7, 1, 6, 9, 4};
    int tamanho = sizeof(array) / sizeof(array[0]);
    int valorProcurado = 7;

    int indiceEncontrado = buscaSequencial(array, tamanho, valorProcurado);

    if (indiceEncontrado != -1) {
        cout << "Valor " << valorProcurado << " encontrado no índice: " << indiceEncontrado << endl;
    } else {
        cout << "Valor " << valorProcurado << " não encontrado no array." << endl;
    }

    return 0;
}
